dialer.go

```diff
@@ -170,6 +170,21 @@ func NewDialer(ctx context.Context, opts ...Option) (*Dialer, error) {
 	return d, nil
 }
 
+// verifyIAMAuthNConfig checks that the engine support automatic IAM authn. If
+// auto IAM authn was not request, this is a no-op.
+func verifyIAMAuthNConfig(version string, wantsIAMAuthN bool) error {
+	// Don't check engine version if IAM AuthN isn't requested.
+	if !wantsIAMAuthN {
+		return nil
+	}
```

Comment:
nit: this should be up a level (so you don't need to pass in wantsIAMAuth

Then you can rename the function (supportsAutoIAM) which is a bit more clear